targets/TARGET_Samsung/TARGET_SIDK_S5JS100/serial_api.c

```diff
@@ -0,0 +1,570 @@
+
+static const PinMap PinMap_UART_TX[] = {
+	{UART_TX0, UART_0, 0},
+	{UART_TX1, UART_1, 0},
+	{UART_TX2, UART_2, 0},
+	{UART_TX3, UART_3, 0},
+	{NC, NC, 0}
+};
+static const PinMap PinMap_UART_RX[] = {
+	{UART_RX0, UART_0, 0},
+	{UART_RX1, UART_1, 0},
+	{UART_RX2, UART_2, 0},
+	{UART_RX3, UART_3, 0},
+	{NC, NC, 0}
+};
+#define UART_NUM 4
+
+static uart_irq_handler irq_handler[UART_NUM];
+
+int stdio_uart_inited = 0;
+serial_t stdio_uart;
+
+struct serial_global_data_s {
+	uint32_t serial_irq_id;
+	gpio_t sw_rts, sw_cts;
+	uint8_t count, rx_irq_set_flow, rx_irq_set_api;
+};
+
+static struct serial_global_data_s uart_data[UART_NUM];
+
+void serial_init(serial_t *obj, PinName tx, PinName rx)
+{
+	int is_stdio_uart = 0;
+
+	// determine the UART to use
+	UARTName uart_tx = (UARTName)pinmap_peripheral(tx, PinMap_UART_TX);
+	UARTName uart_rx = (UARTName)pinmap_peripheral(rx, PinMap_UART_RX);
+	UARTName uart = (UARTName)pinmap_merge(uart_tx, uart_rx);
+	if ((int)uart == NC) {
+		error("Serial pinout mapping failed");
+	}
+
+	obj->uart = (void *)uart;
+
+	// make sure every pending clear
+	USI_PTR(uart)->UINTM = 0xF;
+	USI_PTR(uart)->UINTP = 0xF;
+
+	//set baud rate and enable Uart in normarl mode (RX and TX enabled)
+	switch (uart) {
+	case UART_0: 
+		S5JS100_USI0_UART->UFCON = UART_UFCON_TX_FIFO_RESET; // reset FIFO not to break terminal
+		wait_ms(10);
+		S5JS100_USI0_UART->UCON = UART_UCON_RX_TOUT_32FRAMES |
+								  UART_UCON_TX_INTTYPE_LEVEL | UART_UCON_RX_INTTYPE_LEVEL |
+								  UART_UCON_RX_TOUTINT_ENABLE | UART_UCON_RX_ERRINT_DISABLE |
+								  UART_UCON_TX_MODE_IRQPOLL | UART_UCON_RX_MODE_IRQPOLL;
+		S5JS100_USI0_UART->UFCON = 0x227; // revise me
+		S5JS100_USI0_UART->UINTM = UART_UINTM_MODEM_MASK |
+									UART_UINTM_TXD_MASK |
+								   UART_UINTM_RXD_MASK; // revise me */
+
+		/*   S5JS100_USI0_UART->UFTL_CONF = 0x0;  */
+		break;
+	case UART_1: 
+#if 1	
+		/* temporal setting for USI_UART1 */
+		////HACK///////////////////////////////////
+		putreg32(1, 0x830150C4); /* Reset USI1 */
+		putreg32(1, 0x85041038); /* Set USI1 as UART*/
+		putreg32(0, 0x830150C4); /* Release USI1 */
+
+		putreg32(0x7301, 0x82021130); /* Set GPI12 as USI1_RXD UART */
+		putreg32(0x7301, 0x82021134); /* Set GPI13 as USI1_TXD UART */
+		S5JS100_USI1_UART->UFCON =  UART_UFCON_TX_FIFO_RESET | UART_UFCON_RX_FIFO_RESET; // reset FIFO not to break terminal
+		S5JS100_USI1_UART->UFCON =  UART_UFCON_FIFO_DISABLE;
+#if 1
+		/* USI_UART1 is used */
+		*(volatile unsigned int *)(0x83015000) = 0x3;
+		*(volatile unsigned int *)(0x83015004) = 0x3385;
+		*(volatile unsigned int *)(0x83015008) = 0x221;
+		*(volatile unsigned int *)(0x8301500C) = 0x0;
+
+		*(volatile unsigned int *)(0x83015010) = 0x6;
+		*(volatile unsigned int *)(0x83015014) = 0x0;
+		*(volatile unsigned int *)(0x83015018) = 0x0;
+		*(volatile unsigned int *)(0x8301501C) = 0x1;
+
+	/*  *(volatile unsigned int *)(0x83014020) = 0x0;  */
```

Comment:
please remove the dead code

Reply:
also on lines below, lines 209, 186 and others

Please check the rest of the files as well